File: m/rls/wscon.r
file	wscon - console application
include	rid:wsdef
include	rid:cldef
include	rid:imdef
include	rid:mxdef
include	rid:medef
include	rid:stdef
include	<stdio.h>
	main : (int, **char) int extern

;	Handles console applications 

  func	WinMain
	ins : HINSTANCE
	prv : HINSTANCE
	cmd : LPSTR
	sho : int
	()  : APIENTRY int
  is	buf : [512] char
	cnt : int = 0
	vec : ** char = <>
	st_cop ("image ", buf)			; dummy image name
	st_app (cmd, buf)			; the command
	cl_vec (buf, &cnt, &vec)		; leave space for image name
	main (cnt, vec)				;
	im_exi ()				;
  end
end file

code	ws_prt - printf replacement

;	Just bypass our renaming mechanism

  func	ws_prt
	fmt : * char const
	... : ...
  	()  : int
  is	vprintf(fmt, <va_list>&(fmt) + #fmt)
	fine
  end

code	ws_msg - write message

  func	ws_msg
	msg : * char
	obj : * char
  is	reply im_con (msg, obj)		; bypass the bypass
  end


File: m/rls/rlb/wscon.c
/* file -  wscon - console application */
#include "m:\rid\wsdef.h"
#include "m:\rid\cldef.h"
#include "m:\rid\imdef.h"
#include "m:\rid\mxdef.h"
#include "m:\rid\medef.h"
#include "m:\rid\stdef.h"
#include <stdio.h>
extern int main (int ,char **);
int APIENTRY WinMain(
HANDLE ins ,
HANDLE prv ,
LPSTR cmd ,
int sho )
{ char buf [512];
  int cnt = 0;
  char *(*vec )= NULL;
  st_cop ("image ", buf);
  st_app (cmd, buf);
  cl_vec (buf, &cnt, &vec);
  main (cnt, vec);
  im_exi ();
} 
/* code -  ws_prt - printf replacement */
int ws_prt(
const char *fmt ,
...)
{ vprintf(fmt, (va_list )&(fmt) +  sizeof(fmt));
  return 1;
} 
/* code -  ws_msg - write message */
ws_msg(
char *msg ,
char *obj )
{ PUT ("%s%s-", imPpre, imPfac);
  PUT (msg, obj);
  PUT ("\n");
  return 1;
} 
